pkg/i2gw/providers/common/utils.go

```diff
@@ -17,24 +17,104 @@ limitations under the License.
 package common
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"log"
 	"regexp"
+	"strings"
 
 	networkingv1 "k8s.io/api/networking/v1"
 	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
 )
 
-func GetIngressClass(ingress networkingv1.Ingress) string {
-	var ingressClass string
+var k8sClientset *kubernetes.Clientset
+
+func init() {
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(errors.New("unable to create config"))
+	}
+
+	k8sClientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func getFileExtension(fileName string) string {
+	parts := strings.Split(fileName, ".")
+	if len(parts) > 1 {
+		return parts[len(parts)-1]
+	}
+	return ""
+}
+
+func readIngressClassesFromFile(data []byte, extension string) (*networkingv1.IngressClassList, error) {
+	ingressClasses := &networkingv1.IngressClassList{}
+	var err error
+
+	switch extension {
+	case "yaml":
+		err = yaml.Unmarshal(data, ingressClasses)
+	case "json":
+		err = json.Unmarshal(data, ingressClasses)
+	default:
+		return nil, fmt.Errorf("unsupported file type: %s", extension)
+	}
```

Comment:
Maybe I missed something but it seems like you're always sending an empty slice of data